### PR TITLE
pycbc_live: check state & DQ of followup detectors

### PR DESCRIPTION
**Pull request by titodalcanton**
Adds a check for the state and DQ of followup detectors when calculating the SNR timeseries for PyCBC Live. Needs testing, but I am opening for consideration. In particular I want to make two notes:

* A followup detector is ignored if it was offline or had DQ issues at any point during the duration of the template, with some padding and taking into account the duration of the SNR timeseries. Maybe this is too strict though. I.e. if DQ issues affect just a few samples at the beginning or end of the SNR series, there may still be value in sending the timeseries to BAYESTAR. Sounds like some risk/benefit tradeoff needs to be discussed. For the moment it might be ok to be overly conservative, and maybe revisit the decision once Virgo proves to be super valuable for sky localization.
* Detectors which made the coincidence (i.e. not followup detectors) are assumed to have valid data for the whole SNR timeseries and are *not* checked. In principle they can be checked as well, but I am not sure what to do if the check fails! In that case we claim to have a coincidence, but we refuse to report the SNR timeseries around it, which sounds strange.

Thoughts?

### CODE REVIEW

**Comment by ahnitz** [comment]
For the *key* detectors, we do do some check around the time of the trigger with a padding of +- 1 s if I remember correctly, so it will mostly be OK, I think. Issues in the early inspiral for BNS for example, I think are less worrying concerns than the problems caused by more vanilla glitches. The description of the changes sounds like the right approach to me.

File: pycbc/frame/frame.py [hunk 1]
```diff
@@ -631,7 +631,8 @@ def __init__(self, frame_src,
                        max_buffer=2048,
                        valid_mask=3,
                        force_update_cache=False,
-                       increment_update_cache=None):
+                       increment_update_cache=None,
+                       padding=0):
```
Comment (ahnitz) [inline pycbc/frame/frame.py]: Hmmm, maybe just keep this in the LiveStrainBuffer class? If we aren't using it here, it seems a bit odd to me.

Comment (ahnitz) [inline pycbc/frame/frame.py]: i.e. access later as data_buffer[ifo].dq_padding instead of dq.padding.

**Comment by ahnitz** [comment]
@titodalcanton BTW, can you do a test upload to gracedb to see if bayestar is responding correctly and the page looks sane?

**Comment by titodalcanton** [comment]
I am testing offline now and once things look sane, I will.

**Comment by titodalcanton** [comment]
Here are plots showing the results with offline simulations (same as used for the poster):
https://www.atlas.aei.uni-hannover.de/~tito/LSC/pycbc/bayestar_subthreshold/series_check_%231722/
Compare to
https://www.atlas.aei.uni-hannover.de/~tito/LSC/pycbc/bayestar_subthreshold/series_check_H1L1_edgecases/
Note that these do not actually test the state/DQ feature though.

**Pull request merged**